lib/plugins/create/templates/twilio-runtime/handler.js

```diff
@@ -0,0 +1,11 @@
+const logo = require('asciiart-logo');
```

Comment:
Where is this dependency coming from? Shouldn't this be listed in the `package.json` dependencies section? Or is this available in the Twilio Runtime? 🤔

Reply:
it's defined in the `serverless.yml`. 

https://github.com/serverless/serverless/pull/6467/files#diff-c9fa620e15d789a64e35aeb136e52e5bR16-R17

I didn't want to mix this with the dependencies defined in the `package.json`. Several other parts of our tooling define the function dependencies in a separated config. For the case of serverless framework I thought this makes sense to have all the config in the `serverless.yml`. WDYT?

Reply:
Interesting. Haven't seen that 🤔

Thanks for clarifying :+1:

The other provider plugins are implemented in a way that one has to explicitly define them in `package.json` (or whatever runtime / package manager you're using).

I know that this is not well documented but we suggest to put everything proprietary (which is not part of the original `serverless.yml` spec) under the `custom` property. This way the Framework can be extended alter on without causing name collision.

TBH I'm fine with either way (whatever is best for Twilio), but I'd suggest to move this property to `custom`:

```yaml
custom:
  dependencies:
    asciiart-logo: '*'
```

Reply:
Thanks for having another look.  @pmuens 

Your points are very interesting and bring up an additional question on my end. I had a look at the initial AWS `serverless.yml` and it turns out my proposal is different not only in the dependency part.

I defined the following different properties:

- `provider.environment` - which has a completely different meaning then in the AWS config (the reason for that is that in the twilio runtime `environment` is similar to what a `stage` in AWS is – I decided to follow naming conventions in our docs)
- `provider.config` - needed configuration token (I more or less copied that approach from the cloudflare serverless plugin implementation 🙈)
- `provider.environmentVars` - the environment variables available in the service
- `provider.dependencies` - needed npm deps which will be provided

I feel like it will be tough to match the AWS keys/properties because there are some major differences and not everything is mappable.

About the `custom` approach – I don't have strong feeling about this but I'm not sure if that's not more confusing to users to have important things in a not related `custom` property. Dependency definitions, auth credentials and environment configuration belongs on the provider level for me. 

I also looked at the other providers and it feels like there a many differences already. :)

----- 

Long story short – it feels like I won't be able to match every needed config in Twilio Runtime to AWS properties and am not sure about the `custom` property. 🙈 I'm more than happy to rename keys but for now would prefer to have everything in the `provider` (if it's really cool for you?).

Reply:
Friendly ping @pmuens. 😊

As far as I see this is the missing part to move forward with this PR. :)

Reply:
Hey @stefanjudis,
sorry for the slow response on my end. This somehow slipped through the cracks, so thanks for pinging me :+1:

I agree after reading your comment here. Since this is a provider plugin it's fine to have the important config parts at the `provider` level. I furthermore agree that putting things into different places will just cause confusion, so the choices you made here are absolutely right 👌